package.json

```diff
@@ -124,10 +124,11 @@
     "pino": "7.6.0",
     "prismjs": "1.25.0",
     "react": "17.0.2",
+    "react-async-hook": "4.0.0",
```

Comment:
This introduces the useAsync hook. Why add another library to do async fetch when we already have react-query ?

Reply:
This isn't used for data fetching. Some of the atoms replaced returned promises, so this library handles their resolution to expose the resolved response (the same as before)

Reply:
To elaborate, react query is really for revalidating stale data, using the swr pattern. 

In this PR, I turn an async atom into a hook, such as the ones that generate txs, which need to be resolved. ReactQuery _could_ do this, but it's not really it's job. This library is tiny, and is just for resolving promises.

Reply:
Just food for thought: https://tkdodo.eu/blog/react-query-as-a-state-manager

> React Query is an async state manager. It can manage any form of asynchronous state - it is happy as long as it gets a Promise.

Reply:
A useful article, but it still talks about state that needs to synchronises with a backend. The use here is different: generating the unsigned tx. For example, with the same input: it never goes stale, never needs to rerender on mount, or on page focus (all of which would need to be disabled trying to use react-query).

The API of this lib is way more simple: if any of these dependencies change, reresolve the promise, just as a `useEffect` would rerun on a dep change. 

The readme [discusses the difference between this lib and a swr-style libs](https://github.com/slorber/react-async-hook#things-we-dont-support-by-design).

Reply:
Thanks for the detailed clarification. Sounds good to me ! This lib is indeed pretty small, has no dependency (except React) and does one job well 👍.